File: worksheets/dp-four/scala/src/visitor/HtmlElement.scala
package visitor

case class HtmlElement(var tagName: String) extends HtmlTag {
  var startTag: String = ""
  var endTag: String = ""

  private var tagBody: String = ""

  override def setTagBody(tagBody: String): Unit = ???
  override def generateHtml(): Unit = ???
  override def accept(visitor: Visitor): Unit = ???

  def getEndTag(): String = ???
  def getStartTag(): String = ???
  def getTagName(): String = ???
  def setEndTag(tag: String): Unit = ???
  def setStartTag(tag: String): Unit = ???
}

File: worksheets/dp-five/kotlin/src/visitor/HtmlElement.kt
package visitor

data class HtmlElement(val startTag: String,
                       val tagBody: String,
                       val endTag: String) : HtmlTag(startTag, endTag) {

    override fun generateHtml() {
        TODO()
    }

    override fun accept(visitor: Visitor) {
        TODO()
    }

    override fun addChildTag(htmlTag: HtmlTag) {
        TODO("not implemented")
    }

    override fun removeChildTag(htmlTag: HtmlTag) {
        TODO("not implemented")
    }

    override fun getChildren(): List<HtmlTag> {
        TODO("not implemented")
    }

}
